test/e2e/framework/node/wait.go

```diff
@@ -0,0 +1,197 @@
+			if !IsConditionSetAsExpected(&node, v1.NodeReady, true) {
+				notReady = append(notReady, node)
+			}
+		}
+		pods, err := c.CoreV1().Pods(metav1.NamespaceAll).List(metav1.ListOptions{ResourceVersion: "0"})
+		if err != nil {
+			return false, err
+		}
+
+		systemPodsPerNode := make(map[string][]string)
+		for _, pod := range pods.Items {
+			if pod.Namespace == metav1.NamespaceSystem && pod.Status.Phase == v1.PodRunning {
+				if pod.Spec.NodeName != "" {
+					systemPodsPerNode[pod.Spec.NodeName] = append(systemPodsPerNode[pod.Spec.NodeName], pod.Name)
+				}
+			}
+		}
+		missingPodsPerNode = make(map[string][]string)
+		for _, node := range nodes.Items {
+			if !system.IsMasterNode(node.Name) {
+				for _, requiredPod := range requiredPerNodePods {
+					foundRequired := false
+					for _, presentPod := range systemPodsPerNode[node.Name] {
+						if requiredPod.MatchString(presentPod) {
+							foundRequired = true
+							break
+						}
+					}
+					if !foundRequired {
+						missingPodsPerNode[node.Name] = append(missingPodsPerNode[node.Name], requiredPod.String())
+					}
+				}
+			}
+		}
+		return len(notReady) == 0 && len(missingPodsPerNode) == 0, nil
+	})
+
+	if err != nil && err != wait.ErrWaitTimeout {
+		return err
+	}
+
+	if len(notReady) > 0 {
+		return fmt.Errorf("Not ready nodes: %v", notReady)
+	}
+	if len(missingPodsPerNode) > 0 {
+		return fmt.Errorf("Not running system Pods: %v", missingPodsPerNode)
+	}
+	return nil
+
+}
+
+// WaitConditionToBe returns whether node "name's" condition state matches wantTrue
+// within timeout. If wantTrue is true, it will ensure the node condition status
+// is ConditionTrue; if it's false, it ensures the node condition is in any state
+// other than ConditionTrue (e.g. not true or unknown).
+func WaitConditionToBe(c clientset.Interface, name string, conditionType v1.NodeConditionType, wantTrue bool, timeout time.Duration) bool {
+	e2elog.Logf("Waiting up to %v for node %s condition %s to be %t", timeout, name, conditionType, wantTrue)
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(poll) {
+		node, err := c.CoreV1().Nodes().Get(name, metav1.GetOptions{})
+		if err != nil {
+			e2elog.Logf("Couldn't get node %s", name)
+			continue
+		}
+
+		if IsConditionSetAsExpected(node, conditionType, wantTrue) {
+			return true
+		}
+	}
+	e2elog.Logf("Node %s didn't reach desired %s condition status (%t) within %v", name, conditionType, wantTrue, timeout)
+	return false
+}
+
+// WaitForNameToBeNotReady returns whether node name is not ready (i.e. the
+// readiness condition is anything but ready, e.g false or unknown) within
+// timeout.
+func WaitForNameToBeNotReady(c clientset.Interface, name string, timeout time.Duration) bool {
+	return WaitConditionToBe(c, name, v1.NodeReady, false, timeout)
+}
+
+// WaitForNameToBeReady returns whether node name is ready within timeout.
+func WaitForNameToBeReady(c clientset.Interface, name string, timeout time.Duration) bool {
```

Comment:
Here you change the signature from `WaitForNodeToBeReady` to `WaitForNameToBeReady`, yes?
wouldn't it read better `e2enode.WaitForNodeToBeReady`? 
I think the name makes more sense given what it does.

Reply:
same as above

Reply:
For this one, `e2enode.WaitForNameToBeReady` sounds a bit weird to me, whereas `e2enode.WaitForNodeToBeReady` tells me exactly what it does. I think keeping the extra "node" in the method signature adds value (is not purely redundant).

Reply:
will fix, thanks